src/app/ui/components/input/input.tsx

```diff
@@ -0,0 +1,206 @@
+import {
+  type ComponentProps,
+  LegacyRef,
+  createContext,
+  forwardRef,
+  useContext,
+  useImperativeHandle,
+  useRef,
+  useState,
+} from 'react';
+
+import { sva } from 'leather-styles/css';
+import { SystemStyleObject } from 'leather-styles/types';
+
+import { useOnMount } from '@app/common/hooks/use-on-mount';
+import { propIfDefined } from '@app/common/utils';
```

Comment:
Maybe `propIfDefined` should be under `ui/utils` as that's the only place we are using it and its technically part of the UI library? 

Mentioning to open the discussion on how we separate we want to keep things